ouroboros-consensus/docs/VersioningSchemeDecision.md

```diff
@@ -0,0 +1,577 @@
+- Remove all files from the three `wip-status/{major,minor,patch}` directories.
+
+_Remark_.
+If you maintain a separate changelog per package, then `scriv`'s [Categories](https://github.com/nedbat/scriv/blob/cd79a2a618eb752075bbf45e93ad6b7576a2924a/docs/concepts.rst#categories) can be leveraged for this, instead of these empty files.
+
+Pros:
+
+- A PR's contribution to the subsequent release's version bump is judged when reviewing the PR instead of being assessed (many days) later.
+
+- It does not have distinct dev version and release versions, but inspecting the contents of a specific commit would let you determine whether its actually the release version it declares or else some evolution of it.
+  But that requires inspecting the special files, which users, downstream devs, and standard tools won't do.
+  (However, see the Remark below.)
+
+Cons:
+
+- It requires the additional consideration/discussion of which of the three `wip-status/{major,minor,patch}` directories to add the empty file in for each package the PR touches.
+  This makes it a little harder to merge a PR, but not as much as writing the changelog entry would.
+  (It avoids merge/rebase conflicts the same way that `scriv` does.)
+
+_Remark_.
+You could also add the extra `.0` version component from Proposal Redimensional in a commit immediately _after_ the release.
+That extra `.0` would be naturally removed by the pre-release commit, since you're only releasing a package if its `wip-status` directory is non-empty.
+It suffers the same coarseness of dev revisions as Proposal Redimensional, because every commit between release `A.B.C.D` and release `A.B.(C+1).0` would be versioned `A.B.C.*.0` even if some have more features than the other---and similar for major version bumps and breaking changes.
+But that should be mostly harmless, since development versions (`A.B.C.D.0`) should be visible downstream only when developers are very explicitly opting-in to them.
+
+# Second Draft of this whole document
+
+When should you update the version declarations in your package manifests (eg the top-level `version:` field in your `.cabal` file)?
+We would prefer for some unassailable authority to enumerate and explain the archetypal options, but we have not yet discovered such an article in our research or discussions.
+This short document therefore assembles what prior art we have found into such a resource.
+
+Executive summary: the challenge of maintaining versions is qualitatively equivalent to the challenge of maintaining a changelog.
+
+### Design Space Overview
+
+It's a surprisingly rich design space, but the following concepts delineate its perimeter and the main trends within it.
+
+A principal dichotomy among possible version declaration maintenance schemes has become apparent during our research.
+- Definition _RisingEdge_. The version declaration is altered by (some of the) development PRs; it's already correct when it comes time to announce a release.
```

Comment:
These definitions are reversed compared to the previous draft! Which perhaps suggests they're not that obvious as names :)

Reply:
Yes, I realized that same thing last night!

And agree that it means they're not good names.